### PR TITLE
add more deposit cases

### PR DESCRIPTION
**Pull request by cloud8little**
Add serveral negative test scenarios include the following (not limited):

- one of the proofs is incorrect.
- contract insuffcient to send.
- one of the signatures is incorrect.
- to address is contract address

### CODE REVIEW

**Comment by cloud8little** [comment]
merge? @mialbu

File: test/DepositTest.ts [hunk 1]
```diff
@@ -111,16 +112,12 @@ describe("Bridge contract", function () {
...
-
       const proofs = [proof1];
       const msgToSign = concatRoots(proofs);
       const signatures = await getValidatorSignatures(msgToSign);
 
-      await bridgeContract.connect(relayer).deposit(proofs, signatures);
-      expect(await ethers.provider.getBalance(to1)).to.equal(balanceBeforeTo1 + toEthDecimals(proof1.amount));
-      expect(await ethers.provider.getBalance(await bridgeContract.getAddress())).to.equal(balanceBeforeBridge - toEthDecimals(proof1.amount));
+      const tx = await bridgeContract.connect(relayer).deposit(proofs, signatures);
+      await expect(tx).to.changeEtherBalances([bridgeContract, to1], [-toEthDecimals(proof1.amount), toEthDecimals(proof1.amount)]);
```
Comment (mialbu) [inline test/DepositTest.ts:120]: oh great, didn't know this assertion exists :100:

File: test/DepositTest.ts [hunk 2]
```diff
@@ -133,40 +130,103 @@ describe("Bridge contract", function () {
...
+      const to = await bridgeContract.getAddress()
+      await relayer.sendTransaction({ to, value: amount });
+
+      const proofs = [proof1, proof2];
+      const msgToSign = concatRoots(proofs);
+      const signatures = await getValidatorSignatures(msgToSign);
+      // Depends on What happens if the transfer fails?
+      const tx = await bridgeContract.connect(relayer).deposit(proofs, signatures);
+      await expect(tx).to.changeEtherBalances([bridgeContract, to1, to2], [-toEthDecimals(proof1.amount), toEthDecimals(proof1.amount), 0]);
+      // await expect(bridgeContract.connect(relayer).deposit(proofs,signatures)).to.be.revertedWithPanic(0x1);
```
Comment (mialbu) [inline test/DepositTest.ts]: I've discussed this with Hu and Steven, and we'll introduce a claim method. If a transfer fails (i.e., if `transfer` returns `false`) we'll add that deposit to a claim map that will allow claiming it individually at a later time.
I've pushed a commit that lets this test fail, and we can make it pass once the claim is implemented.

File: test/DepositTest.ts [hunk 3]
```diff
@@ -133,40 +130,103 @@ describe("Bridge contract", function () {
...
+      fundContract(bridgeContract, relayer);
+
+      const bridgeContractAddress = await bridgeContract.getAddress();
+      const { proof, root } = await createMerkleTree(2, bridgeContractAddress, ethers.parseEther("1"));
+      const self_createproof = { nonce: 2, to: bridgeContractAddress, amount: ethers.parseEther("1"), proof: proof, root: root };
+      const proofs = [proof1, self_createproof];
+      const msgToSign = concatRoots(proofs);
+      const signatures = await getValidatorSignatures(msgToSign);
+      const tx = await bridgeContract.connect(relayer).deposit(proofs, signatures);
+      await expect(tx).to.changeEtherBalances([bridgeContract, proof1.to], [-toEthDecimals(proof1.amount), toEthDecimals(proof1.amount)]);
```
Comment (mialbu) [inline test/DepositTest.ts]: Same as above, I've added a commit to make this test fail for now until the claim functionality is introduced.

**Review by mialbu — APPROVED** [review]

**Pull request merged**